### DISCUSSION TITLE
Docs for "Separate OpenAPI schemas for input and output" should be updated because of a Pydantic change

### DISCUSSION THREAD

**Discussion by RomeoDespres** [fastapi/fastapi#10453] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# This code is taken from the documentation but is incomplete.
# See fix in the description

from pydantic import BaseModel


class Item(BaseModel):   
    name: str
    description: str | None = None
```

### Description

In the docs on separate OpenAPI schemas for input and output ([here](https://fastapi.tiangolo.com/how-to/separate-openapi-schemas/)) it is implied that using pydantic v2 will by default enable the separate schema behavior.

However since pydantic v2.4.0, defaults values are serialized as non-required by default (see [release notes](https://docs.pydantic.dev/latest/changelog/#v240-2023-09-22)).

To enable the schema separation behavior, the following configuration must now be added:

```python
from pydantic import BaseModel, ConfigDict


class Item(BaseModel):
    model_config = ConfigDict(json_schema_serialization_defaults_required=True)

    name: str
    description: str | None = None

```

Shouldn't this be mentioned in FastAPI docs? I've just spent half a day trying to understand why my schemas were *not* separate for input and output, thought others may face that too

### Operating System

Windows

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.11.1

**Comment by CroMarmot** [comment]
fastapi 0.111.0
pydantic 2.7.1
python3 3.10.12

maybe relative: https://github.com/hey-api/openapi-ts/discussions/597
